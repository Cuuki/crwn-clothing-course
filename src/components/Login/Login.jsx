import React, {useState, useCallback} from 'react';

import Button from '../UI/Button';
import FormInput from '../UI/FormInput';

import './Login.scss';

const Login = ({googleSignInStart, credentialSignInStart}) => {
  const [userCredentials, setUserCredentials] = useState({
    email: '',
    password: '',
  });
  const {email, password} = userCredentials;

  const handleSubmit = useCallback(
    (event) => {
      event.preventDefault();

      credentialSignInStart(email, password);
    },
    [credentialSignInStart, email, password]
  );

  const handleChange = useCallback(
    (event) => {
      const {value, name} = event.target;

      setUserCredentials({
        ...userCredentials,
        [name]: value,
      });
    },
    [userCredentials]
  );

  return (
    <div className="login">
      <h2 className="title">I already have an account</h2>
      <span>Sign in with your email and password</span>
      <form action="" onSubmit={handleSubmit}>
        <FormInput
          id="loginEmail"
          type="email"
          name="email"
          value={email}
          label="Email"
          required="required"
          onChange={handleChange}
        />
        <FormInput
          id="loginPassword"
          type="password"
          name="password"
          value={password}
          label="Password"
          required="required"
          onChange={handleChange}
        />
        <div className="buttons">
          <Button type="submit">Login</Button>
          <Button type="button" onClick={googleSignInStart} isGoogleSignIn>
            Login with Google
          </Button>
        </div>
      </form>
    </div>
  );
};

export default Login;
