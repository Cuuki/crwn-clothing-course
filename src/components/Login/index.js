
import {connect} from 'react-redux';
import {
  googleSignInStart,
  credentialSignInStart,
} from '../../actions/userActions';

import Login from './Login';

const mapDispatchToProps = (dispatch) => ({
  googleSignInStart: () => dispatch(googleSignInStart()),
  credentialSignInStart: (email, password) =>
    dispatch(credentialSignInStart({email, password})),
});

export default connect(null, mapDispatchToProps)(Login);
