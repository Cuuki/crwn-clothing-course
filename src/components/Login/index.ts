import {connect} from 'react-redux';
import {
  credentialSignInStart,
  googleSignInStart,
} from '../../redux/modules/user';
import Login from './Login';

const mapDispatchToProps = (dispatch) => ({
  googleSignInStart: () => dispatch(googleSignInStart()),
  credentialSignInStart: (email, password) =>
    dispatch(credentialSignInStart({email, password})),
});

export default connect(null, mapDispatchToProps)(Login);
