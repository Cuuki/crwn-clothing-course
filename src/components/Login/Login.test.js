import React from 'react';
import {render} from "@testing-library/react";
import Login from './Login';

describe('Login', () => {
  it('renders Login component', () => {
    const mockGoogleSignInStart = jest.fn();
    const mockCredentialSignInStart = jest.fn();

    expect(render(
      <Login
        googleSignInStart={mockGoogleSignInStart}
        credentialSignInStart={mockCredentialSignInStart}
      />
    )).toMatchSnapshot();
  });
})
