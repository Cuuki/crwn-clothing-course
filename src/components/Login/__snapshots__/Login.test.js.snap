// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders Login component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login"
      >
        <h2
          class="title"
        >
          I already have an account
        </h2>
        <span>
          Sign in with your email and password
        </span>
        <form
          action=""
        >
          <div
            class="group"
          >
            <input
              class="form-input"
              name="email"
              required=""
              type="email"
            />
            <label
              class="form-input-label"
              for="loginEmail"
            >
              Email
            </label>
          </div>
          <div
            class="group"
          >
            <input
              class="form-input"
              name="password"
              required=""
              type="password"
            />
            <label
              class="form-input-label"
              for="loginPassword"
            >
              Password
            </label>
          </div>
          <div
            class="buttons"
          >
            <button
              class="sc-bdnylx XiWmD custom-button"
              type="submit"
            >
              Login
            </button>
            <button
              class="sc-bdnylx kLurem custom-button"
              type="button"
            >
              Login with Google
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login"
    >
      <h2
        class="title"
      >
        I already have an account
      </h2>
      <span>
        Sign in with your email and password
      </span>
      <form
        action=""
      >
        <div
          class="group"
        >
          <input
            class="form-input"
            name="email"
            required=""
            type="email"
          />
          <label
            class="form-input-label"
            for="loginEmail"
          >
            Email
          </label>
        </div>
        <div
          class="group"
        >
          <input
            class="form-input"
            name="password"
            required=""
            type="password"
          />
          <label
            class="form-input-label"
            for="loginPassword"
          >
            Password
          </label>
        </div>
        <div
          class="buttons"
        >
          <button
            class="sc-bdnylx XiWmD custom-button"
            type="submit"
          >
            Login
          </button>
          <button
            class="sc-bdnylx kLurem custom-button"
            type="button"
          >
            Login with Google
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
